include/posix/sys/types.h

```diff
@@ -85,6 +88,8 @@ typedef struct pthread_rwlock_obj {
 
 #endif /* CONFIG_PTHREAD_IPC */
 
+#endif /* CONFIG_ARCH_POSIX */
```

Comment:
This is not nice, as you just remove that type and therefore that API when compiling for the posix arch